[python-updates] Recurse into python313Packages

mweinelt opened the pull request:
Early prep for the 25.05 release cycle.

- Recurse into python313Packages
- Stop recursing into python311Packages
- Update the default numpy version

mweinelt committed:
python311Packages: stop recursing into package set

For NixOS 24.11 we are going to support Python 3.12/3.13.

mweinelt committed:
python313Packages: recurse into package set

Python 3.313 is the latest stable Python release and will be supported
through NixOS 24.11 and 25.05.

mweinelt committed:
python313Packages.colorlog: fix build

mweinelt committed:
python313Packages.jedi: 0.19.1 -> 0.19.1-unstable-2024-10-17

Fast-forward to git master, which is supposed to get released as 0.19.2 soon.

mweinelt committed:
python313Packages.rpds-py: 0.18.1 -> 0.20.0

mweinelt committed:
python313Packages.wrapt: 1.16.0 > 1.17.0dev4

https://github.com/GrahamDumpleton/wrapt/compare/refs/tags/1.16.0...1.17.0dev4

mweinelt committed:
python313Packages.curio: 1.6 -> 1.6-unstable-2024-04-11

mweinelt committed:
python313Packages.parameterized: fix tests

Kudos to the Gentoo folks for the patch.

mweinelt committed:
python313Packages.trustme: 1.1.0 -> 1.2.0

https://trustme.readthedocs.io/en/latest/#trustme-1-2-0-2024-10-07

mweinelt committed:
python313Packages.watchdog: 4.0.2 -> 5.0.3

https://github.com/gorakhargosh/watchdog/blob/v5.0.3/changelog.rst

mweinelt committed:
python313Packages.pure-eval: 0.2.2 -> 0.2.3

https://github.com/alexmojaki/pure_eval/compare/v0.2.2...v0.2.3

mweinelt committed:
python313Packages.astroid: 3.3.4 -> 3.3.5

https://github.com/PyCQA/astroid/blob/refs/tags/v3.3.5/ChangeLog

mweinelt committed:
python313Packages.twisted: 24.7.0 -> 24.10.0

https://github.com/twisted/twisted/blob/twisted-24.10.0/NEWS.rst

mweinelt committed:
python313Packages.cython_0: disable

mweinelt committed:
python313Packages.pyflakes: disable failing tests

mweinelt committed:
python313Packages.oauthlib: disable failing test

Python 3.13 started dedenting multiline strings, which made this test fail.

mweinelt committed:
gpgme: add python3.13 support

mweinelt committed:
python3.pkgs.pythonRuntimeDepsCheckHook: allow prereleases

Updating to prereleases should be possible, but making this an option is
difficult, given that for packages with many consumers you would have to
set it in each consumer.

We thoroughly test the package set, so allowing prereleases unconditionally
shouldn't be too bad.

Closes: #301698

mweinelt committed:
python313Packages.toolz: 0.12.1 -> 1.0.0

https://github.com/pytoolz/toolz/releases/tag/1.0.0

mweinelt committed:
python313Packages.pytest-doctestplus: fix tests

mweinelt committed:
python312Packages.cython_0: 0.29.36 -> 0.29.37.1

https://github.com/cython/cython/blob/0.29.37.1/CHANGES.rst

mweinelt committed:
python313Packages.psycopg2: 2.9.9 -> 2.9.10

natsukium committed:
python312Packages.numpy: 1.26.4 -> 2.1.2

mweinelt committed:
python312Packages.redis: 5.1.1 -> 5.2.0

https://github.com/redis/redis-py/releases/tag/v5.2.0

mweinelt committed:
python312Packages.fakeredis: 2.25.1 -> 2.26.1

https://github.com/cunla/fakeredis-py/releases/tag/v2.26.1

natsukium committed:
python312Packages.seaborn: fix compatibility with numpy>=2

natsukium committed:
python312Packages.cirq-core: 1.4.1 -> 1.4.1-unstable-2024-09-21

fix compatibility with numpy>=2

natsukium committed:
python312Packages.ase: fix compatibility with numpy>=2

mweinelt committed:
python313Packages.watchfiles: 0.22.0 -> 0.24.0

https://github.com/samuelcolvin/watchfiles/compare/refs/tags/v0.22.0...v0.24.0

mweinelt committed:
python313Packages.httptools: 0.6.1 -> 0.6.4

https://github.com/MagicStack/httptools/releases/tag/v0.6.2
https://github.com/MagicStack/httptools/releases/tag/v0.6.3
https://github.com/MagicStack/httptools/releases/tag/v0.6.4

mweinelt committed:
python313Packages.deprecated: disable failing tests

and use the PEP517 builder.

natsukium committed:
python312Packages.einops: fix compatibility with numpy>=2

mweinelt committed:
python313Packages.flit: 3.9.0 -> 3.10.0

https://github.com/pypa/flit/blob/3.10.0/doc/history.rst

natsukium committed:
python312Packages.patsy: fix compatibility with numpy>=2

natsukium committed:
python312Packages.plotly: fix compatibility with numpy>=2

mweinelt committed:
python312Packages.setuptools: 75.1.0 -> 75.3.0

https://setuptools.pypa.io/en/stable/history.html#v75-3-0

natsukium commented:
Regarding numpy2, when will we update it? Since python313 package set adopts numpy2 by default, I think it would be a good idea to bump it here to avoid confusion.

I have a few fixes and here is my local hydra evaluation as of about a week ago.
I'm a little concerned about jax, but the rest looks good.

Diff:
https://github.com/NixOS/nixpkgs/compare/master...natsukium:nixpkgs:numpy2

Hydra evaluation report:
https://gist.github.com/natsukium/e4362b577d41ac8f8b7962446c0c4c44

mweinelt commented:
> Regarding numpy2, when will we update it? Since python313 package set adopts numpy2 by default, I think it would be a good idea to bump it here to avoid confusion.

Let's give it a shot. I said we'd delay it until after the release, and here we are.

> I'm a little concerned about jax, but the rest looks good.

Too early to worry imo.

mweinelt committed:
python3.pkgs.pythonRuntimeDepsCheckHook: don't validate wheel metadata

Metadata can have issues that the packaging library will complain about,
but that should not prevent us from building any package.

mweinelt committed:
python313Packages.packaging: 24.1 -> 24.2

https://github.com/pypa/packaging/blob/24.2/CHANGELOG.rst

mweinelt committed:
python312Packages.starlette: 0.40.0 -> 0.41.2

https://www.starlette.io/release-notes/#0412

mweinelt committed:
python312Packages.python-multipart: 0.0.12 -> 0.0.17

https://github.com/Kludex/python-multipart/releases/tag/0.0.13
https://github.com/Kludex/python-multipart/releases/tag/0.0.14
https://github.com/Kludex/python-multipart/releases/tag/0.0.15
https://github.com/Kludex/python-multipart/releases/tag/0.0.16
https://github.com/Kludex/python-multipart/releases/tag/0.0.17

mweinelt committed:
python312Packages.fastapi: 0.115.3 -> 0.115.4

https://github.com/fastapi/fastapi/releases/tag/0.115.4

mweinelt committed:
python313Packages.eliot: patch for numpy2 compat

mweinelt committed:
python313Packages.eventlet: apply patch for 3.13 compat

mweinelt committed:
python313Packages.msgspec: apply patch for 3.13 support

mweinelt committed:
python312Packages.moto: 5.0.18 -> 5.0.20

https://github.com/getmoto/moto/blob/5.0.20/CHANGELOG.md

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/524.html

mweinelt committed:
python313Packages.orjson: 3.10.7 -> 3.10.11

https://github.com/ijl/orjson/blob/3.10.11/CHANGELOG.md

mweinelt committed:
python313Packages.numpy: 2.1.2 -> 2.1.3

https://github.com/numpy/numpy/releases/tag/v2.1.3

mweinelt committed:
python313Packages.ipdb: disable failing tests

mweinelt committed:
python313Packages.pkginfo: 1.11.1 -> 1.11.2

mweinelt committed:
python313Packages.pyodbc: 5.1.0 -> 5.2.0

https://github.com/mkleehammer/pyodbc/releases/tag/5.2.0

mweinelt committed:
python313Packagese.cattrs: disable failing tests, modernize

mweinelt committed:
python313Packages.ml-dtypes: establish numpy2 compat

mweinelt commented:
Okay, so jax fails a number of tests (the same on 3.12 and 3.13), but we can still try to update it.

```
FAILED tests/debugger_test.py::CliDebuggerTest::test_debugger_can_print_backtrace - jaxlib.xla_extension.XlaRuntimeError: INTERNAL: Generated function failed: ...
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUnique1 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUnique2 - AssertionError: Tuples differ: (2,) != (2, 1, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUnique4 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize0 - AssertionError: Tuples differ: (4,) != (1, 4)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize4 - AssertionError: Tuples differ: (4,) != (1, 4)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize5 - AssertionError: Tuples differ: (1,) != (1, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize6 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize8 - AssertionError: Tuples differ: (4,) != (1, 1, 4)
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testUniqueSize9 - AssertionError: Tuples differ: (3,) != (3, 1)
FAILED tests/lax_numpy_test.py::NumpySignaturesTest::testWrappedSignaturesMatch - AssertionError: {'clip': {'np_params': ['a', 'a_min', 'a_max', 'min', 'max'...
FAILED tests/lax_numpy_reducers_test.py::JaxNumpyReducerTests::testCumulativeSum0 - AssertionError:
```

mweinelt committed:
python312Packages.pytest-benchmark: 4.0.0 -> 5.1.0

https://github.com/ionelmc/pytest-benchmark/blob/v5.1.0/CHANGELOG.rst

dotlambda committed:
python312Packages.baize: init at 0.22.2

dotlambda committed:
python312Packages.a2wsgi: fix tests

dotlambda committed:
python312Packages.connexion: skip failing test

mweinelt committed:
python313Packages.django_4: disable failing test

mweinelt committed:
python313Packages.importlib-resources: backport 3.13 compat

mweinelt committed:
python313Packages.hatchling: 1.25.0 -> 1.26.1

https://github.com/pypa/hatch/releases/tag/hatchling-v1.26.0
https://github.com/pypa/hatch/releases/tag/hatchling-v1.26.1

mweinelt committed:
python313Packages.aiohappyeyeballs: 2.4.2 -> 2.4.3

https://github.com/bdraco/aiohappyeyeballs/blob/v2.4.3/CHANGELOG.md

dotlambda committed:
python312Packages.graphql-core: 3.2.4 -> 3.2.5

Diff: https://github.com/graphql-python/graphql-core/compare/refs/tags/v3.2.4...v3.2.5

Changelog: https://github.com/graphql-python/graphql-core/releases/tag/v3.2.5

mweinelt committed:
python313Packages.sphinx-autodoc-typehints: 2.4.4 -> 2.5.0

https://github.com/tox-dev/sphinx-autodoc-typehints/releases/tag/2.5.0

mweinelt committed:
python312Packages.nbmake: create home for ipython tests

mweinelt committed:
python312Packages.datalad: ignore deprecation warning in tests

mweinelt committed:
python313Packages.nbformat: disable failing tests

mweinelt committed:
python312Packages.filterpy: disable failing test

and make the derivation a PEP517 build with the relevant lingo.

mweinelt committed:
python312Packages.norfair: relax numpy constraint

and apply PEP517 lingo.

mweinelt committed:
python313Packages.css-inline: 0.14.1 -> 0.14.2

https://github.com/Stranger6667/css-inline/blob/python-v0.14.2/CHANGELOG.md

mweinelt committed:
python313Packages.pycurl: 7.45.3 -> 7.45.3-unstable-2024-10-17

Update to unstable version to relieve some compat issues and reenable
a number of tests.

The upstream unfortunately does not push out release very often anymore.

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/536.html

mweinelt committed:
python313Packages.pytest-benchmark: disable failing test

mweinelt committed:
python313Packages.sqlalchemy-utils: disable failing test

mweinelt committed:
python312Packages.mxnet: relax numpy constraint

mweinelt committed:
python312Packages.pyvirtualdisplay: test without xdist

High core counts tend to produce hangs and defunct python processes. And
while upstream has put pytest-xdist into its test requirements they don't
actually use it themselves either.

mweinelt committed:
python313Packages.future: disable

No compatibility has been established as of 2024-11.

mweinelt committed:
python313Packages.python-json-logger: disable failing test

mweinelt committed:
python313Packages.python-json-logger: migrate to PEP517 build

mweinelt committed:
python313Packages.coreapi: disable

Consumer of the retired cgi module. The upstream project is also
archived.

mweinelt committed:
python313Packages.djangorestframework: fix build

As coreapi does not support Python 3.13, make its consumption conditional
on the python version.

mweinelt committed:
python313Packages.pytest-codspeed: init at 3.0.0

mweinelt committed:
python313Packages.yarl: 1.13.1 -> 1.17.1

https://github.com/aio-libs/yarl/blob/v1.17.1/CHANGES.rst

mweinelt committed:
python313Packages.aiohttp: 3.10.10 -> 3.11.0

https://github.com/aio-libs/aiohttp/blob/v3.11.0/CHANGES.rst

mweinelt committed:
python313Packages.aiohttp-fast-zlib: disable tests

API changes in aiohttp 3.11.0 caused the test suite to break.

mweinelt committed:
python313Packages.aioresponses: fix compat with aiohttp 3.11.0

mweinelt committed:
python312Packages.elastic-transport: ignore deprecation warnings in tests

mweinelt committed:
python313Packages.aiohttp-fast-zlib: 0.1.1 -> 0.2.0

https://github.com/bdraco/aiohttp-fast-zlib/blob/v0.2.0/CHANGELOG.md

mweinelt committed:
python313Packages.aiogram: 3.13.1 -> 3.14.0

https://github.com/aiogram/aiogram/releases/tag/v3.14.0

mweinelt committed:
python313Packages.aiohttp-zlib-ng: drop

Archived by upstream in favor of aiohttp-fast-zlib and ultimately broke
in the aiohttp 3.11.0 update.

dotlambda committed:
python313Packages.coloredlogs: replace pipes module usage

It was deprecated in 3.11 and removed in 3.13.

dotlambda committed:
python313Packages.coloredlogs: use pep517 builder

dotlambda committed:
python313Packages.pyarrow: use Cython 3

dotlambda commented:
@teh @thoughtpolice @tscholak Can we remove six and future from torch's dependencies?

FliegendeWurst committed:
python3Packages.uvloop: use system libuv (#351870)

Co-Authored-By: Martin Weinelt <hexa@darmstadt.ccc.de>

dotlambda committed:
python313Packages.testfixtures: fix tests

dotlambda committed:
python313Packages.patsy: 0.5.6 -> 1.0.1

Diff: https://github.com/pydata/patsy/compare/refs/tags/v0.5.6...v1.0.1

Changelog: https://github.com/pydata/patsy/releases/tag/v1.0.0
           https://github.com/pydata/patsy/releases/tag/v1.0.1

dotlambda committed:
python313Packages.venusian: fix tests

dotlambda committed:
python313Packages.compreffor: 0.5.5 -> 0.5.6

mweinelt committed:
python313Packages.falcon: 3.1.3 -> 4.0.2

https://falcon.readthedocs.io/en/stable/changes/4.0.0.html
https://falcon.readthedocs.io/en/stable/changes/4.0.1.html
https://falcon.readthedocs.io/en/stable/changes/4.0.2.html

mweinelt committed:
python313Packages.hug: drop

Marked broken since 2021, still pins falcon==2.0.0 and we are at falcon>4
now.

mweinelt committed:
python313Packages.audioop-lts: init at 0.2.1

mweinelt committed:
python313Packages.aiohttp: 3.11.0 -> 3.11.2

https://github.com/aio-libs/aiohttp/blob/v3.11.2/CHANGES.rst

mweinelt committed:
python313Packages.nbconvert: disable DeprecationWarning in tests

mweinelt committed:
python313Packages.pyramid: fix build

mweinelt committed:
python313Packages.tiktoken: 0.7.0 -> 0.8.0

https://github.com/openai/tiktoken/blob/0.8.0/CHANGELOG.md

mweinelt committed:
python313Packages.objgraph: 3.6.1 -> 3.6.2

https://github.com/mgedmin/objgraph/blob/3.6.2/CHANGES.rst

mweinelt committed:
python312Packages.python-lsp-server: disable failing test

mweinelt committed:
python313Packages.sqlalchemy: 2.0.34 -> 2.0.36

https://github.com/sqlalchemy/sqlalchemy/releases/tag/rel_2_0_36

mweinelt committed:
python312Packages.inkex: relax numpy constraint

mweinelt committed:
python313Packages.quantities: 0.15.0 -> 0.16.1

https://github.com/python-quantities/python-quantities/blob/v0.16.1/CHANGES.txt

mweinelt committed:
python312Packages.pydevd: disable failing test

This is due to the numpy 2 migration.

dotlambda committed:
python313Packages.pint: 0.24.3 -> 0.24.4

Diff: https://github.com/hgrecco/pint/compare/refs/tags/0.24.3...0.24.4

Changelog: https://github.com/hgrecco/pint/blob/0.24.4/CHANGES

dotlambda committed:
python313Packages.flexparser: 0.3.1 -> 0.4

Diff: https://github.com/hgrecco/flexparser/compare/0.3.1...0.4

Changelog: https://github.com/hgrecco/flexparser/blob/0.4/CHANGES

mweinelt committed:
python313Packages.quart: 0.19.8 -> 0.19.9

https://github.com/pallets/quart/blob/refs/tags/0.19.9/CHANGES.rst

dotlambda committed:
python313Packages.geojson: fix build

mweinelt committed:
python313Packages.werkzeug: 3.0.6 -> 3.1.3

https://werkzeug.palletsprojects.com/en/3.1.x/changes/#version-3-1-3

mweinelt committed:
python313Packages.blinker: 1.8.2 -> 1.9.0

https://github.com/pallets-eco/blinker/releases/tag/1.9.0

mweinelt committed:
python313Packages.flask: 3.0.3 -> 3.1.0

https://flask.palletsprojects.com/en/3.1.x/changes/#version-3-1-0

mweinelt committed:
octoprint: fix flask override

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/549.html

mweinelt committed:
python312Packages.flask-sqlalchemy: update disabled tests

dotlambda commented on pkgs/applications/misc/octoprint/default.nix:
```diff
@@ -34,6 +34,7 @@ let
           flask = super.flask.overridePythonAttrs (oldAttrs: rec {
             version = "2.2.5";
             format = "setuptools";
+            pyproject = null;
```
Why not override `build-system` instead?

mweinelt replied:
No strong opinion. Was just focused on getting rid of the eval error.

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/551.html

mweinelt committed:
sourcehut: fix flask ecosystem overrides

mweinelt committed:
python313Packages.asn1: fix build

mweinelt committed:
python313Packages.yeelight: fix build

mweinelt committed:
python313Packages.commonmark: revamp and fix build

mweinelt committed:
python313Packages.tree-sitter: 0.23.0 -> 0.23.2

https://github.com/tree-sitter/py-tree-sitter/releases/tag/v0.23.1
https://github.com/tree-sitter/py-tree-sitter/releases/tag/v0.23.2

mweinelt committed:
python313Packages.flask-sqlalchemy: disable tests temporarily

They regressed due to Python 3.13 changes, but that doesn't mean that the
library itself isn't working.

mweinelt committed:
python313Packages.unify: disable

doronbehar commented:
I iterated some of the commits related to packages I have some interest in, and they look good. Not sure if I have time to help further. Sorry.
